agent/engine/docker_task_engine.go

```diff
@@ -612,7 +605,14 @@ func (engine *DockerTaskEngine) StateChangeEvents() chan statechange.Event {
 
 // AddTask starts tracking a task
 func (engine *DockerTaskEngine) AddTask(task *api.Task) error {
-	task.PostUnmarshalTask(engine.cfg, engine.credentialsManager)
+	err := task.PostUnmarshalTask(engine.cfg, engine.credentialsManager, engine.resourceFields)
+	if err != nil {
+		seelog.Errorf("Task engine [%s]: unable to progress task: %v", task.Arn, err)
+		task.SetKnownStatus(api.TaskStopped)
+		task.SetDesiredStatus(api.TaskStopped)
+		engine.emitTaskEvent(task, err.Error())
+		return nil
```

Comment:
I don't know what all the `return nil` statements in this method are achieving (it's been there for ever). Can you modify this method to not return any error?

Reply:
Sure, how about doing it in a different PR? Changes seem to be in a lot of files including tests.

Reply:
I can add a `TODO`.

Reply:
Since this is a refactor PR, I'd much rather do it in this one